### PR TITLE
Fix alert [object Object] login bug

### PR DESCRIPTION
**Pull request by gaoxk**
## Notion ticket link

[Fix alert [object Object] login bug](https://www.notion.so/uwblueprintexecs/a658c933d18c48d4b43fd99bda2bd3b4?v=ba22ac93d09040ff8219c60ac7caabdb&p=324709d7c0fb478d890378e9baa84a87)

## Implementation description

- The google login component triggers it's onFailure function when third party cookies are blocked (ie, in incognito). The function is modified to not make an annoying window alert when used in incognito.

## Steps to test

1. open the web app
2. clear your cookies and refresh
3. note no window alert
4. repeat 1-3 but in incognito

## What should reviewers focus on?

- your happiness 😎 

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/auth/Login.tsx [hunk 1]
```diff
@@ -48,6 +48,20 @@ const Login = () => {
...
+    error: string;
+    details: string;
+  };
+
+  const onFailure = (response: GoogleErrorResponse) => {
+    // https://stackoverflow.com/questions/63631849/google-sign-in-not-working-in-incognito-mode
+    if (response.error === "idpiframe_initialization_failed") {
+      console.warn("Google SignIn does not work on incognito mode.");
+    } else {
+      alert(JSON.stringify(response));
```
Comment (gaoxk) [inline frontend/src/components/auth/Login.tsx:61]: just alerting an object leads to `[object Object]` which gives the user no information. `JSON.stringify(response)` will actually show the full response (eg `{"error":"idpiframe_initialization_failed","details":"Cookies are not enabled in current environment."}`

File: frontend/src/components/auth/Login.tsx [hunk 2]
```diff
@@ -48,6 +48,20 @@ const Login = () => {
...
 
+  type GoogleErrorResponse = {
+    error: string;
+    details: string;
+  };
+
+  const onFailure = (response: GoogleErrorResponse) => {
+    // https://stackoverflow.com/questions/63631849/google-sign-in-not-working-in-incognito-mode
+    if (response.error === "idpiframe_initialization_failed") {
+      console.warn("Google SignIn does not work on incognito mode.");
```
Comment (jennchenn) [inline frontend/src/components/auth/Login.tsx:59]: do we want to show this error to users so they know why they could not log in? (unless it is already, i haven't been able to replicate this error locally)

**Review by jennchenn — APPROVED** [review]
🥳 no more mysterious [object Object] alerts!

**Review by Puepis — APPROVED** [review]
tested it and LGTM!

**Review by jennifer-lu — APPROVED** [review]
Looks good to me too!

**Comment by gaoxk** [comment]
TFTR!!

**Pull request merged**